### PR TITLE
[CognitiveServices] LUIS Programmatic SDK - Updates based on latest Spec

### PR DESCRIPTION
**Pull request by JuanAr**
Updates based on latest Spec changes (Azure/azure-rest-api-specs#2083)
* Documentation updates
* New Operation IDs
* Unit tests for all operations

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
@JuanAr please fix your commit history, either squash or create a new PR
Can you also do the following in the below order:
1)	Pull latest from upstream (if you have already done this good)
2)	Execute msbuild.exe build.proj from the repo root
a.	This will update the build tools in the current repo
3)	Generate your SDK
4)	Build you sdk at least one from command line of IDE
a.	Msbuild.exe build.proj /t:build /p:Scope=SDKs\CognitiveService\management
5)	Update the PR

**Comment by JuanAr** [comment]
Hi @shahabhijeet, 
The PR changelist was squashed and the build process tested as you requested. 
Regards,

**Comment by shahabhijeet** [comment]
@DavidLiCIG can you approve this PR. Also JuanAr has to be part of Azure org on github

File: src/SDKs/CognitiveServices/dataPlane/Language/LUIS-Programmatic/Customizations/ApiKeyServiceClientCredentials.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    /// <summary>
+    /// Allows authentication to the API using a basic apiKey mechanism
+    /// </summary>
+    public class ApiKeyServiceClientCredentials : ServiceClientCredentials
+    {
+        private readonly string subscriptionKey;
+
+        /// <summary>
+        /// Creates a new instance of the ApiKeyServiceClientCredentails class
```
Comment (chrimc62) [inline src/SDKs/CognitiveServices/dataPlane/Language/LUIS-Programmatic/Customizations/ApiKeyServiceClientCredentials.cs]: Documentation standards require punctuation and sentences here and throughout the file.

File: src/SDKs/CognitiveServices/dataPlane/Language/LUIS-Programmatic/Customizations/Models/ErrorResponse.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
+﻿namespace Microsoft.Azure.CognitiveServices.Language.LUIS.Programmatic.Models
+{
+    using Newtonsoft.Json;
+
+    public partial class ErrorResponse
+    {
+        private string code;
+        private string message;
+
+        public string Code
```
Comment (chrimc62) [inline src/SDKs/CognitiveServices/dataPlane/Language/LUIS-Programmatic/Customizations/Models/ErrorResponse.cs]: Missing documentation.

**Review by chrimc62 — APPROVED** [review]
Aside from the minor nit of XML comments needing periods at the end of sentences, this looks pretty good.  I like the extensive tests and the model is consistent with the API.  (Eventually I would prefer URLs not based on GUIDs so everything isn't two round trips.)

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Please talk to David li as I have suggested the directory structure for Language data plane SDK.
If you have any questions, please let me know.

Also please onboard using aka.ms/azuregithub
This seems you are not authorized to kick off CI jobs.
Once you get onboarded the CI job will be kicked off and once the CI passes this PR will be ready to be reviewed.

**Comment by DavidLiCIG** [comment]
Hi @shahabhijeet, this directory structure looks fine and I agree with you on the other email with regards to modular Nuget packages. Let's get this in if it looks good.

**Comment by pcostantini** [comment]
Hi @shahabhijeet  & @DavidLiCIG ,

The directory structure was updated as discussed and latest changes from the Spec added to the Programmatic SDK.

image

**Comment by shahabhijeet** [comment]
@JuanAr closing at request of Pablo

**Pull request closed without merging**